lib/webauthn/encoder.rb

```diff
@@ -1,55 +1,112 @@
 
-  class Encoder
-    # https://www.w3.org/TR/webauthn-2/#base64url-encoding
-    STANDARD_ENCODING = :base64url
+  # Custom error class for encoding-related errors
+  class EncodingError < StandardError; end
 
+  # Handles encoding and decoding of WebAuthn data
+  class Encoder
     attr_reader :encoding
 
-    def initialize(encoding = STANDARD_ENCODING)
+    # Initialize encoder with specified encoding type
+    # @param encoding [Symbol] the encoding type to use (:base64url by default)
+    def initialize(encoding = Config::Encoder::DEFAULT_ENCODING)
       @encoding = encoding
     end
 
+    # Encode data using the specified encoding type
+    # @param data [String] the data to encode
+    # @return [String] encoded data
     def encode(data)
-      case encoding
-      when :base64
-        [data].pack("m0") # Base64.strict_encode64(data)
-      when :base64url
-        data = [data].pack("m0") # Base64.urlsafe_encode64(data, padding: false)
-        data.chomp!("==") or data.chomp!("=")
-        data.tr!("+/", "-_")
-        data
-      when nil, false
-        data
-      else
-        raise "Unsupported or unknown encoding: #{encoding}"
-      end
+      return data if skip_encoding?
+
+      validate_encoding!
+      send("encode_#{encoding}", data)
     end
 
+    # Decode data using the specified encoding type
+    # @param data [String] the data to decode
+    # @return [String] decoded data
     def decode(data)
-      case encoding
-      when :base64
-        data.unpack1("m0") # Base64.strict_decode64(data)
-      when :base64url
-        if !data.end_with?("=") && data.length % 4 != 0 #  Base64.urlsafe_decode64(data)
-          data = data.ljust((data.length + 3) & ~3, "=")
-          data.tr!("-_", "+/")
-        else
-          data = data.tr("-_", "+/")
-        end
-        data.unpack1("m0")
-      when nil, false
-        data
-      else
-        raise "Unsupported or unknown encoding: #{encoding}"
-      end
+      return data if skip_encoding?
+
+      validate_encoding!
+      send("decode_#{encoding}", data)
+    end
+
+    private
+
+    # Check if encoding should be skipped
+    # @return [Boolean] true if encoding is nil or false
+    def skip_encoding?
+      encoding.nil? || encoding == false
+    end
+
+    # Validate that the current encoding type is supported
+    # @raise [EncodingError] if encoding type is not supported
+    def validate_encoding!
+      return if Config::Encoder::ENCODINGS.key?(encoding)
+
+      raise EncodingError, Config::Encoder::INVALID_ENCODING_ERROR % encoding
     end
+
+    # Encode data using standard Base64 encoding
+    # @param data [String] the data to encode
+    # @return [String] Base64 encoded data
+    def encode_base64(data)
+      Base64.strict_encode64(data)
+    end
+
+    # Encode data using URL-safe Base64 encoding without padding
+    # @param data [String] the data to encode
+    # @return [String] URL-safe Base64 encoded data
+    def encode_base64url(data)
```

Comment:
It'd be nice to have these methods public in some way, so that they can be used in our specs to encode/decode data and we can remove the `base64` dependency altogether – see [this PR](https://github.com/cedarcode/webauthn-ruby/pull/423).